lib/index.js

```diff
@@ -1,88 +1,356 @@
+ * @param  {Object}     location   The location object of the CodeClimate issue.
+ * @return {Array[]}               The range: `[[lineNumber, colStart], [lineNumber, colEnd]]`.
+ */
+const calcRange = (textEditor, location) => {
+  // Issue only has a line number
+  if (!Object.prototype.hasOwnProperty.call(location, 'positions')) {
+    return Helpers.generateRange(textEditor, location.lines.begin - 1);
+  }
+
+  const { positions } = location;
+  const line = positions.begin.line - 1;
+
+  // Invalid starting column, just treat it as a line number
+  if (positions.begin.column === undefined) {
+    return Helpers.generateRange(textEditor, line);
+  }
+
+  const colStart = (positions.begin.column - 1) || 0;
+  const colEnd = (positions.end.column === undefined)
+    ? undefined : (positions.end.column - 1);
+
+  // No valid end column, let generateRange highlight a word
+  if (colEnd === undefined || colStart === colEnd) {
+    return Helpers.generateRange(textEditor, line, colStart);
+  }
+
+  // Valid end column, and different from the start one
+  return [[line, colStart], [line, colEnd]];
+};
+
+/**
+ * @summary Fetch the paths of all currently open files on Atom.
+ * @return  {Object} Dictionary of open file abspaths ~> textEditor.
+ *
+ * NOTE Files indexed by abspath to avoid relative filepath collision among different projects.
+ */
+const fetchOpenFilepaths = () => {
+  const openFiles = {};
+  atom.workspace.textEditorRegistry.editors.forEach((textEditor) => {
+    openFiles[textEditor.getPath()] = textEditor;
+  });
+  return openFiles;
+};
+
+/**
+ * @summary Parses the issues reported by CodeClimate CLI to the format AtomLinter expects.
+ * @param   {Object}    path    Object with paths for project root and triggering file.
+ * @param   {Object}    result  JSON string from the CodeClimate CLI output to parse.
+ * @return  {Object[]}          Parsed issues, with following keys per oobject (array item):
+ *                              - severity: the issue severity (one of (info|warning|error)).
+ *                              - excerpt: summary of the issue.
+ *                              - description: explanation of the issue.
+ *                              - location: { file, position }.
+ */
+const parseIssues = (path, result) => {
+  let messages;
+
+  try {
+    messages = JSON.parse(result);
+  } catch (e) {
+    notifyError(e, 'Invalid JSON returned from CodeClimate. See the Console for details.');
+    // eslint-disable-next-line no-console
+    console.error('Invalid JSON returned from CodeClimate:', result);
+    return [];
+  }
+
+  const open = fetchOpenFilepaths();
+  const linterResults = [];
+  messages.forEach((issue) => {
+    // Exit early if not an issue
+    if (issue.type.toLowerCase() !== 'issue') return;
+
+    // Exit early if issued file is not open
+    const file = join(path.project, issue.location.path);
+    if (!open[file]) return;
```

Comment:
I would instead use something like [findTextEditor](https://github.com/AtomLinter/linter-elixirc/blob/987d21dca55131fa061223a0d80b138c30f7d4a1/lib/init.js#L104-L111) to get the `TextEditor` for a file _if one is available_, if not return the best position you can. If the user then opens the file a `TextEditor` will be available and a more accurate position can be determined from that.

Once you start returning results for anything other than the active `TextEditor` that you are passed, you _must_ change the scope to `project`, and since that allows you to send results for _any_ path, there is no reason to only filter results to ones that the user has open.